### DISCUSSION TITLE
Installing and hosting plugins via npm registry

### DISCUSSION THREAD

**Discussion by hanseltime** [yarnpkg/berry#5282] [Ideas]
First off, I love the ability to add plugins to yarn.  It has made very intricate monorepos super easy to augment.  My one issue is that  yarn seems to only provide a reliable package for their plugins and all third party plugins seem to require pointing to a git source url.

From a corporate code perspective, I would much rather be able to point to a package in registry that has a definitive bundled state and doesn't give other developers pause when looking at imports.

Is there a way for this to be done already?  Is it just a matter of yarn not socializing a process to bundle packages like so?  I would love to understand this, because that would help clean up my code substantially.

**Comment by RDIL** [comment]
The problem with this is that the plugins themselves are needed to download things from the registry (using the npm plugin, which is built-in).  Solutions have been proposed in the form of downloading missing plugins, which could probably let you use something like unpkg.

**Comment by hanseltime** [comment]
Yea, I figured as much.  It's almost like there would need to be a "raw loader" and "normal loader", where the raw loader could download most all plugins that don't actively listen to raw load events.  Changing loader lifecycle like that wouldn't be an easier or clean break out I'm guessing.
